src/fsharp/vs/service.fs

```diff
@@ -1635,134 +1634,156 @@ module internal Parser =
+                    // Replay other background errors.
+                    for (phasedError,sev) in otherBackgroundDiagnostics do
+                        if sev = FSharpErrorSeverity.Warning then 
+                            warning phasedError.Exception 
+                        else errorR phasedError.Exception
+            
+                | None -> 
+                    // For non-scripts, check for disallow #r and #load.
+                    ApplyMetaCommandsFromInputToTcConfig tcConfig (parsedMainInput,Path.GetDirectoryName mainInputFileName) |> ignore
+                    
+                // A problem arises with nice name generation, which really should only 
+                // be done in the backend, but is also done in the typechecker for better or worse. 
+                // If we don't do this the NNG accumulates data and we get a memory leak. 
+                tcState.NiceNameGenerator.Reset()
+                
+                // Typecheck the real input.  
+                let sink = TcResultsSinkImpl(tcGlobals, source = source)
+            
+                let! tcEnvAtEndOpt =
+                    async {
+                        try
+                            let checkForErrors() = (parseResults.ParseHadErrors || errHandler.ErrorCount > 0)
+                            // Typecheck is potentially a long running operation. We chop it up here with an Eventually continuation and, at each slice, give a chance
+                            // for the client to claim the result as obsolete and have the typecheck abort.
+                            //let computation = TypeCheckOneInputAndFinishEventually(checkForErrors,tcConfig, tcImports, tcGlobals, None, TcResultsSink.WithSink sink, tcState, parsedMainInput)
+                            //match computation |> Eventually.forceWhile (fun () -> not cancellationToken.IsCancellationRequested) with
+                            //| Some((tcEnvAtEnd,_,typedImplFiles),tcState) -> Some (tcEnvAtEnd, typedImplFiles, tcState)
+                            //| None -> None // Means 'aborted'
+                            let projectDir = Path.GetDirectoryName(projectFileName)
+                            let capturingErrorLogger = CompilationErrorLogger("TypeCheckOneFile", tcConfig)
+                            let errorLogger = GetErrorLoggerFilteringByScopedPragmas(false,GetScopedPragmasForInput(parsedMainInput), capturingErrorLogger)
+                            
+                            let! result = 
+                                TypeCheckOneInputAndFinishEventually(checkForErrors,tcConfig, tcImports, tcGlobals, None, TcResultsSink.WithSink sink, tcState, parsedMainInput)
+                                |> Eventually.repeatedlyProgressUntilDoneOrTimeShareOverOrCanceled
+                                    50L
+                                    cancellationToken
+                                    (fun f -> f())
+                                |> Eventually.forceAsync 
```

Comment:
It will (though if the time slices are much smaller than the 50ms then won't checking the cancellation token result in cancellation happening sooner?)